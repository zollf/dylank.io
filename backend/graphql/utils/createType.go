package utils

import "github.com/graphql-go/graphql"

var _cachedListType map[string]*graphql.List
var _cachedObjectType map[string]*graphql.Object

// Creates a cached type, useful for wrapper graphql functions that return type
func CreateList(key string, interfaceConfig *graphql.List) *graphql.List {
	if _cachedListType == nil {
		_cachedListType = make(map[string]*graphql.List)
	}

	if config, ok := _cachedListType[key]; ok {
		return config
	} else {
		_cachedListType[key] = interfaceConfig
		return _cachedListType[key]
	}
}

func CreateObject(key string, interfaceConfig *graphql.Object) *graphql.Object {
	if _cachedObjectType == nil {
		_cachedObjectType = make(map[string]*graphql.Object)
	}

	if config, ok := _cachedObjectType[key]; ok {
		return config
	} else {
		_cachedObjectType[key] = interfaceConfig
		return _cachedObjectType[key]
	}
}
