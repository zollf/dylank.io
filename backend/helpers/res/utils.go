package res

import (
	"fmt"

	"github.com/kataras/iris/v12"
)

func Redirect(ctx iris.Context, params string) bool {
	if redirect := ctx.FormValue("redirect"); redirect != "" {
		ctx.Redirect(fmt.Sprintf("%s%s", redirect, params))
		return true
	}
	return false
}

func AcceptsJSON(ctx iris.Context) bool {
	return ctx.GetHeader("Accepts") == "application/json"
}
