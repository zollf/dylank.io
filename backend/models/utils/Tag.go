package utils

import (
	"app/models/projects"
	"app/models/tags"
	"sort"
	"time"
)

type TagInterface struct {
	ID        uint64    `json:"id"`
	Slug      string    `json:"slug"`
	Title     string    `json:"title"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
	Count     int       `json:"count"`
}

type TagData struct {
	ID        uint64
	Index     int
	Title     string
	Slug      string
	CreatedAt string
	UpdatedAt string
}

type ProjectTagData struct {
	Title   string
	Slug    string
	Checked bool
}

// Given a tag, check it exists amongst an array of tags
func CheckTagExistsInTags(needle *tags.Tag, haystack []*tags.Tag) bool {
	for _, needleInHay := range haystack {
		if needleInHay.ID == needle.ID {
			return true
		}
	}
	return false
}

func TagOccurrencesInProjects(projects []*projects.Project, tag *tags.Tag) int {
	count := 0
	for _, project := range projects {
		for _, projectTag := range project.Tags {
			if projectTag.ID == tag.ID {
				count = count + 1
			}
		}
	}

	return count
}

func TagsOccurrencesInProjects(projects []*projects.Project) []*TagInterface {
	var tags []*TagInterface
	tagsMap := make(map[string]*TagInterface)
	for _, project := range projects {
		for _, tag := range project.Tags {
			if tagInterface, ok := tagsMap[tag.Slug]; ok {
				tagInterface.Count = tagInterface.Count + 1
			} else {
				tagsMap[tag.Slug] = &TagInterface{
					ID:        tag.ID,
					Slug:      tag.Slug,
					Title:     tag.Title,
					CreatedAt: tag.CreatedAt,
					UpdatedAt: tag.UpdatedAt,
					Count:     1,
				}
			}
		}
	}

	keys := make([]string, 0, len(tagsMap))
	for key := range tagsMap {
		keys = append(keys, key)
	}
	sort.Strings(keys)

	for _, key := range keys {
		tags = append(tags, tagsMap[key])
	}

	return tags
}

func GetTagsData() ([]*TagData, error) {
	tags, t_err := tags.All()
	if t_err != nil {
		return nil, t_err
	}

	var tag_data []*TagData
	zone, _ := time.LoadLocation("Australia/Perth")

	for i, tag := range tags {
		tag_data = append(tag_data, &TagData{
			ID:        tag.ID,
			Index:     i + 1,
			Title:     tag.Title,
			Slug:      tag.Slug,
			CreatedAt: tag.CreatedAt.In(zone).Format(time.RFC822),
			UpdatedAt: tag.UpdatedAt.In(zone).Format(time.RFC822),
		})
	}

	return tag_data, nil
}

func GetTagData(id string) (*TagData, error) {
	tag, not_found := tags.Find(id)

	if not_found != nil {
		return nil, not_found
	}

	tag_data := &TagData{
		ID:    tag.ID,
		Title: tag.Title,
		Slug:  tag.Slug,
	}

	return tag_data, nil
}
