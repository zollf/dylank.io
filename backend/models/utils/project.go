package utils

import (
	"app/models/assets"
	"app/models/projects"
	"app/models/tags"
	"time"
)

// Used for dom rendering
type ProjectData struct {
	ID          uint64
	Index       int
	Title       string
	Description string
	CreatedAt   string
	UpdatedAt   string
}

type ProjectDataWithTagAndAsset struct {
	ID          uint64
	Title       string
	Slug        string
	Description string
	Assets      []*assets.Asset
	URL         string
	Git         string
	Tags        []*tags.Tag
	CreatedAt   string
	UpdatedAt   string
}

func GetProjectData() ([]*ProjectData, error) {
	projects, projects_err := projects.All()
	if projects_err != nil {
		return nil, projects_err
	}

	var project_data []*ProjectData
	zone, _ := time.LoadLocation("Australia/Perth")

	for i, project := range projects {
		project_data = append(project_data, &ProjectData{
			ID:          project.ID,
			Index:       i + 1,
			Title:       project.Title,
			Description: project.Description,
			CreatedAt:   project.CreatedAt.In(zone).Format(time.RFC822),
			UpdatedAt:   project.UpdatedAt.In(zone).Format(time.RFC822),
		})
	}

	return project_data, nil
}

func GetProjectDataWithTagAndAsset(id string) (*ProjectDataWithTagAndAsset, []*ProjectTagData, []*ProjectAssetData, error) {
	project, project_err := projects.Find(id)

	if project_err != nil {
		return nil, nil, nil, project_err
	}

	zone, _ := time.LoadLocation("Australia/Perth")

	Git := ""
	if project.Git != nil {
		Git = *project.Git
	}

	URL := ""
	if project.URL != nil {
		URL = *project.URL
	}

	project_data := &ProjectDataWithTagAndAsset{
		ID:          project.ID,
		Title:       project.Title,
		Slug:        project.Slug,
		Assets:      project.Assets,
		URL:         URL,
		Git:         Git,
		Tags:        project.Tags,
		Description: project.Description,
		CreatedAt:   project.CreatedAt.In(zone).Format(time.RFC822),
	}

	var tags_data []*ProjectTagData
	tags, _ := tags.All()
	for _, tag := range tags {
		tags_data = append(tags_data, &ProjectTagData{
			Title:   tag.Title,
			Slug:    tag.Slug,
			Checked: CheckTagExistsInTags(tag, project.Tags),
		})
	}

	var assets_data []*ProjectAssetData
	assets, _ := assets.All()
	for _, asset := range assets {
		assets_data = append(assets_data, &ProjectAssetData{
			Title:   asset.Title,
			Slug:    asset.Slug,
			Url:     asset.Url,
			Checked: CheckAssetExistsInAssets(asset, project.Assets),
		})
	}

	return project_data, tags_data, assets_data, nil
}
