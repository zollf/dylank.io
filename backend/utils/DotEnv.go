package utils

import (
	"fmt"
	"log"

	"github.com/joho/godotenv"
)

func DotEnv() {
	root := RootDir()
	err := godotenv.Load(fmt.Sprintf("%s/.env", root))
	log.Printf("Initialising dotenv")

	if err != nil {
		log.Printf("Failed to initialising dotenv")
	}
}
