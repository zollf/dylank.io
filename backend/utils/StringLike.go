package utils

func StringLike(str string) *string {
	var prtStr *string
	prtStr = &str
	return prtStr
}
