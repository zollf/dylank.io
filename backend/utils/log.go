package utils

import (
	"os"

	"github.com/sirupsen/logrus"
)

var _log *logrus.Logger

func Log() *logrus.Logger {
	if _log == nil {
		_log = logrus.New()
		_log.SetFormatter(&logrus.TextFormatter{
			ForceColors:   true,
			FullTimestamp: true,
		})
		_log.SetOutput(os.Stdout)
	}
	return _log
}
