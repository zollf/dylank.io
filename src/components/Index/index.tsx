import React from 'react';
import useIsMobile from '@/hooks/useIsMobile';

import Header from '../layout/Header';
import Hero from '../Hero';
import MobileHeader from '../layout/MobileHeader';
import styles from './styles.module.scss';

const Index = () => {
  const [isMobile] = useIsMobile(768);
  return (
    <div className={styles.index}>
      {isMobile ? <MobileHeader /> : <Header />}
      <Hero />
    </div>
  );
};

export default Index;
