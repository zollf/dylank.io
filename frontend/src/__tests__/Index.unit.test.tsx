import Index from '../pages/index';
import React from 'react';
import { render } from '@testing-library/react';
import { setMobile } from '@/tests/utils';

it('matches its snapshot', () => {
  const { baseElement } = render(<Index />);
  expect(baseElement).toMatchSnapshot();
});

it('matches its mobile snapshot', () => {
  setMobile();
  const { baseElement } = render(<Index />);
  expect(baseElement).toMatchSnapshot();
});
