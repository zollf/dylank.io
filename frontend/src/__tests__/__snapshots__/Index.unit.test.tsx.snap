// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches its mobile snapshot 1`] = `
<body>
  <div>
    <div
      class="index"
    >
      <div
        class="inner"
      >
        <header
          class="mobileHeader"
        >
          <div
            class="logo"
          >
            d.io
          </div>
          <button
            class="menu"
            data-active="false"
            data-testid="hbtn"
            type="button"
          >
            <test-file-stub />
          </button>
          <div
            class="mobileMenu"
            data-active="false"
            data-testid="mobileMenu"
          >
            <div
              class="inner"
            >
              <nav>
                <a
                  href="#home"
                >
                  Home
                </a>
                <a
                  href="#work"
                >
                  Work
                </a>
                <a
                  href="#about"
                >
                  About
                </a>
                <a
                  href="#contact"
                >
                  Contact
                </a>
              </nav>
              <div
                class="socials"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24 12C24 18.6274 18.6274 24 12 24C5.37258 24 0 18.6274 0 12C0 5.37258 5.37258 0 12 0C18.6274 0 24 5.37258 24 12Z"
                    fill="#7289DA"
                  />
                  <path
                    d="M16.135 6.59998H7.86524C7.16787 6.59998 6.6001 7.15198 6.6001 7.83598V15.948C6.6001 16.632 7.16787 17.184 7.86524 17.184H14.8636L14.5366 16.074L15.3265 16.788L16.0732 17.46L17.4001 18.6V7.83598C17.4001 7.15198 16.8323 6.59998 16.135 6.59998ZM13.7528 14.436C13.7528 14.436 13.5306 14.178 13.3455 13.95C14.1539 13.728 14.4625 13.236 14.4625 13.236C14.2095 13.398 13.9688 13.512 13.7528 13.59C13.4442 13.716 13.148 13.8 12.8579 13.848C12.2655 13.956 11.7224 13.926 11.2595 13.842C10.9078 13.776 10.6054 13.68 10.3523 13.584C10.2104 13.53 10.0561 13.464 9.90181 13.38C9.8833 13.368 9.86478 13.362 9.84627 13.35C9.83393 13.344 9.82775 13.338 9.82158 13.332C9.7105 13.272 9.64878 13.23 9.64878 13.23C9.64878 13.23 9.94501 13.71 10.7288 13.938C10.5436 14.166 10.3153 14.436 10.3153 14.436C8.95141 14.394 8.43301 13.524 8.43301 13.524C8.43301 11.592 9.3217 10.026 9.3217 10.026C10.2104 9.37798 11.0559 9.39598 11.0559 9.39598L11.1176 9.46798C10.0067 9.77998 9.4945 10.254 9.4945 10.254C9.4945 10.254 9.63027 10.182 9.85861 10.08C10.519 9.79798 11.0435 9.71998 11.2595 9.70198C11.2966 9.69598 11.3274 9.68998 11.3644 9.68998C11.7409 9.64198 12.1667 9.62998 12.6111 9.67798C13.1974 9.74398 13.8268 9.91198 14.4687 10.254C14.4687 10.254 13.9811 9.80398 12.932 9.49198L13.0184 9.39598C13.0184 9.39598 13.8639 9.37798 14.7526 10.026C14.7526 10.026 15.6412 11.592 15.6412 13.524C15.6412 13.524 15.1167 14.394 13.7528 14.436ZM10.8831 11.634C10.5313 11.634 10.2536 11.934 10.2536 12.3C10.2536 12.666 10.5375 12.966 10.8831 12.966C11.2348 12.966 11.5126 12.666 11.5126 12.3C11.5187 11.934 11.2348 11.634 10.8831 11.634ZM13.1356 11.634C12.7839 11.634 12.5062 11.934 12.5062 12.3C12.5062 12.666 12.79 12.966 13.1356 12.966C13.4874 12.966 13.7651 12.666 13.7651 12.3C13.7651 11.934 13.4874 11.634 13.1356 11.634Z"
                    fill="white"
                  />
                </svg>
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.0001 0C5.3735 0 0 5.50847 0 12.3038C0 17.74 3.43839 22.3519 8.2064 23.9789C8.80613 24.0927 9.02631 23.7119 9.02631 23.387C9.02631 23.0936 9.01518 22.1244 9.01001 21.0963C5.67157 21.8405 4.96712 19.6446 4.96712 19.6446C4.42125 18.2224 3.63473 17.8443 3.63473 17.8443C2.54596 17.0806 3.7168 17.0963 3.7168 17.0963C4.92181 17.1831 5.55632 18.3642 5.55632 18.3642C6.6266 20.2452 8.36359 19.7014 9.04836 19.387C9.15607 18.5918 9.46706 18.049 9.81024 17.7418C7.14486 17.4307 4.34295 16.3757 4.34295 11.6612C4.34295 10.3179 4.81172 9.22032 5.57937 8.35868C5.45477 8.04878 5.04402 6.79737 5.69562 5.10261C5.69562 5.10261 6.70331 4.77193 8.9965 6.3638C9.95372 6.09119 10.9803 5.95447 12.0001 5.94979C13.0199 5.95447 14.0473 6.09119 15.0063 6.3638C17.2967 4.77193 18.303 5.10261 18.303 5.10261C18.9562 6.79737 18.5452 8.04878 18.4206 8.35868C19.1901 9.22032 19.6557 10.3179 19.6557 11.6612C19.6557 16.3869 16.8484 17.4274 14.1762 17.732C14.6067 18.1138 14.9902 18.8626 14.9902 20.0105C14.9902 21.6568 14.9763 22.9817 14.9763 23.387C14.9763 23.7144 15.1923 24.098 15.8006 23.9772C20.566 22.3485 24 17.7381 24 12.3038C24 5.50847 18.6273 0 12.0001 0Z"
                    fill="#161514"
                  />
                  <path
                    d="M4.58055 17.5812C4.55417 17.6417 4.46055 17.6599 4.37526 17.6183C4.28838 17.5786 4.23959 17.4961 4.26775 17.4354C4.29354 17.373 4.38736 17.3557 4.47403 17.3974C4.56111 17.4372 4.6107 17.5205 4.58055 17.5812Z"
                    fill="#161514"
                  />
                  <path
                    d="M5.06565 18.1315C5.00852 18.1854 4.89685 18.1603 4.82108 18.0752C4.74274 17.9903 4.72806 17.8767 4.78598 17.822C4.84489 17.7682 4.95318 17.7934 5.03173 17.8783C5.11008 17.9643 5.12535 18.077 5.06565 18.1315Z"
                    fill="#161514"
                  />
                  <path
                    d="M5.53775 18.8331C5.46436 18.8849 5.34436 18.8363 5.27018 18.728C5.19679 18.6197 5.19679 18.4897 5.27176 18.4377C5.34614 18.3856 5.46436 18.4324 5.53953 18.54C5.61272 18.6501 5.61272 18.78 5.53775 18.8331Z"
                    fill="#161514"
                  />
                  <path
                    d="M6.18488 19.511C6.11923 19.5846 5.97939 19.5648 5.87705 19.4644C5.77232 19.3661 5.74316 19.2267 5.80901 19.1531C5.87546 19.0792 6.01609 19.1 6.11923 19.1997C6.22316 19.2977 6.2549 19.4381 6.18488 19.511Z"
                    fill="#161514"
                  />
                  <path
                    d="M7.07745 19.9045C7.04849 19.9999 6.91381 20.0433 6.77814 20.0028C6.64267 19.961 6.55401 19.8492 6.58138 19.7528C6.60954 19.6568 6.74482 19.6116 6.88148 19.655C7.01675 19.6965 7.10561 19.8075 7.07745 19.9045Z"
                    fill="#161514"
                  />
                  <path
                    d="M8.05768 19.9776C8.06105 20.0781 7.94601 20.1614 7.8036 20.1632C7.66039 20.1664 7.54456 20.0851 7.54297 19.9863C7.54297 19.8848 7.65543 19.8023 7.79864 19.7999C7.94105 19.7971 8.05768 19.8777 8.05768 19.9776Z"
                    fill="#161514"
                  />
                  <path
                    d="M8.96991 19.8196C8.98697 19.9176 8.888 20.0183 8.74657 20.0451C8.60753 20.071 8.47881 20.0104 8.46115 19.9132C8.4439 19.8127 8.54466 19.7121 8.6835 19.686C8.82512 19.661 8.95186 19.7199 8.96991 19.8196Z"
                    fill="#161514"
                  />
                </svg>
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 12C0 5.37258 5.37258 0 12 0C18.6274 0 24 5.37258 24 12C24 18.6274 18.6274 24 12 24C5.37258 24 0 18.6274 0 12Z"
                    fill="#0077B5"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M8.65948 7.41135C8.65948 8.1959 8.06893 8.82367 7.12069 8.82367H7.10331C6.19033 8.82367 5.6001 8.1959 5.6001 7.41135C5.6001 6.6102 6.2083 6 7.13861 6C8.06893 6 8.64183 6.6102 8.65948 7.41135ZM8.48041 9.93891V18.1098H5.76095V9.93891H8.48041ZM18.2877 18.1098L18.2878 13.4248C18.2878 10.9151 16.9462 9.74704 15.1567 9.74704C13.7129 9.74704 13.0664 10.5401 12.7055 11.0965V9.93913H9.9857C10.0215 10.7058 9.9857 18.11 9.9857 18.11H12.7055V13.5467C12.7055 13.3025 12.7231 13.0589 12.795 12.8841C12.9916 12.3962 13.4391 11.8911 14.1904 11.8911C15.1749 11.8911 15.5684 12.6403 15.5684 13.7384V18.1098H18.2877Z"
                    fill="white"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </div>
        </header>
        <div
          class="hero"
        >
          <div
            class="background"
          >
            <test-file-stub />
          </div>
          <div
            class="content"
          >
            <h2>
              software developer
            </h2>
            <h1>
              Hi, I'm Dylan
            </h1>
            <div
              class="btns"
            >
              <a
                class="button full blue large"
                href="#contact"
                role="button"
                tabindex="0"
              >
                Contact
              </a>
              <a
                class="button full white large"
                href="#work"
                role="button"
                tabindex="0"
              >
                Explore
              </a>
            </div>
          </div>
        </div>
        <div
          class="work"
        >
          <div
            class="top"
          >
            <h1>
              Explore My Projects
            </h1>
          </div>
          <div
            class="tagsWrapper"
          >
            <button
              class="prev hidden"
            >
              <svg
                fill="none"
                height="11"
                viewBox="0 0 7 11"
                width="7"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.5 1L1 5.25L5.5 9.5"
                  stroke="#9CA3AF"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </button>
            <div
              class="tags"
            >
              <button
                class="tag hasFunc active"
                type="button"
              >
                All (
                0
                )
              </button>
              <div
                class="spinner loading"
              >
                <span
                  class="css-119qjy3"
                >
                  <svg
                    class="css-ozkbo8"
                    height="24"
                    viewBox="0 0 16 16"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="8"
                      cy="8"
                      r="7"
                    />
                  </svg>
                </span>
              </div>
            </div>
            <button
              class="next hidden"
            >
              <svg
                fill="none"
                height="11"
                viewBox="0 0 7 11"
                width="7"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 9.5L5.5 5.25L1 1"
                  stroke="#9CA3AF"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </button>
          </div>
          <div
            class="projects"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`matches its snapshot 1`] = `
<body>
  <div>
    <div
      class="index"
    >
      <div
        class="inner"
      >
        <header
          class="header"
        >
          <div
            class="inner"
          >
            <div
              class="logo"
            >
              d.io
            </div>
            <nav>
              <a
                href="#home"
              >
                Home
              </a>
              <a
                href="#work"
              >
                Work
              </a>
              <a
                href="#about"
              >
                About
              </a>
              <a
                href="#contact"
              >
                Contact
              </a>
            </nav>
            <div
              class="socials"
            >
              <a
                href="#"
              >
                <test-file-stub />
              </a>
              <a
                href="#"
              >
                <test-file-stub />
              </a>
              <a
                href="#"
              >
                <test-file-stub />
              </a>
            </div>
          </div>
        </header>
        <div
          class="hero"
        >
          <div
            class="background"
          >
            <test-file-stub />
          </div>
          <div
            class="content"
          >
            <h2>
              software developer
            </h2>
            <h1>
              Hi, I'm Dylan
            </h1>
            <p>
              Expertise in everything web related software. From frontend to backend, infrastructure that's concrete and beautifully displayed. Studying at the University of Western Australia in both Computer Science and Data Science. Have experience in working with big tech stacks and cooperating in teams for huge projects.
            </p>
            <div
              class="btns"
            >
              <a
                class="button blue large"
                href="#contact"
                role="button"
                tabindex="0"
              >
                Contact
              </a>
              <a
                class="button white large"
                href="#work"
                role="button"
                tabindex="0"
              >
                Explore
              </a>
            </div>
          </div>
          <div
            class="bubble1"
          >
            <test-file-stub />
          </div>
          <div
            class="bubble2"
          >
            <test-file-stub />
          </div>
          <div
            class="bubble3"
          >
            <test-file-stub />
          </div>
          <div
            class="bubble4"
          >
            <test-file-stub />
          </div>
        </div>
        <div
          class="work"
        >
          <div
            class="top"
          >
            <h1>
              Explore My Projects
            </h1>
            <p>
              Check my commercial and non commercial projects. If you have any questions feel free to ask me for more information.
            </p>
          </div>
          <div
            class="tagsWrapper"
          >
            <button
              class="prev hidden"
            >
              <svg
                fill="none"
                height="11"
                viewBox="0 0 7 11"
                width="7"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.5 1L1 5.25L5.5 9.5"
                  stroke="#9CA3AF"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </button>
            <div
              class="tags"
            >
              <button
                class="tag hasFunc active"
                type="button"
              >
                All (
                0
                )
              </button>
              <div
                class="spinner loading"
              >
                <span
                  class="css-119qjy3"
                >
                  <svg
                    class="css-ozkbo8"
                    height="24"
                    viewBox="0 0 16 16"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="8"
                      cy="8"
                      r="7"
                    />
                  </svg>
                </span>
              </div>
            </div>
            <button
              class="next hidden"
            >
              <svg
                fill="none"
                height="11"
                viewBox="0 0 7 11"
                width="7"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 9.5L5.5 5.25L1 1"
                  stroke="#9CA3AF"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </button>
          </div>
          <div
            class="projects"
          />
          <div
            class="paginationWrapper"
          >
            <ul
              class="pagination"
            >
              <li
                class="paginationPrev paginationDisabled"
              >
                <a
                  aria-disabled="true"
                  aria-label="Previous page"
                  class=" "
                  rel="prev"
                  role="button"
                  tabindex="-1"
                >
                  <svg
                    fill="none"
                    height="11"
                    viewBox="0 0 7 11"
                    width="7"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.5 1L1 5.25L5.5 9.5"
                      stroke="#9CA3AF"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1.5"
                    />
                  </svg>
                </a>
              </li>
              <li
                class="paginationPage paginationActive"
              >
                <a
                  aria-current="page"
                  aria-label="Page 1 is your current page"
                  rel="canonical"
                  role="button"
                  tabindex="-1"
                >
                  1
                </a>
              </li>
              <li
                class="paginationNext paginationDisabled"
              >
                <a
                  aria-disabled="true"
                  aria-label="Next page"
                  class=" "
                  rel="next"
                  role="button"
                  tabindex="-1"
                >
                  <svg
                    fill="none"
                    height="11"
                    viewBox="0 0 7 11"
                    width="7"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1 9.5L5.5 5.25L1 1"
                      stroke="#9CA3AF"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1.5"
                    />
                  </svg>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
